pkg/server/server_test.go

```diff
@@ -237,3 +242,29 @@ func TestHookAllowsGoodUserGoodDomainWithHTTPSProtocol(t *testing.T) {
 		t.Error("Failed to allow pipelinerun should have been allowed!")
 	}
 }
+
+func TestHookDoesNotAllowUnvettedParameters(t *testing.T) {
+	goodUser := "gooduser"
+	goodDomain := "gooddomain"
+	goodBuilder := "https://good/builder:v1"
+	nsc := &PipelineRunAdmission{
+		AllowedDomains:  []string{goodDomain},
+		AllowedBuilders: []string{goodBuilder},
+		SystemUsers:     []string{goodUser},
+	}
+	server := httptest.NewServer(GetAdmissionServerNoSSL(nsc, ":8080").Handler)
+	badAdmissionReview := getAdmissionReview(goodUser, goodBuilder, goodDomain, []map[string]string{
+		{
+			"name":  "INVALID_PARAM",
+			"value": "not valid",
+		},
+	})
+	requestString := string(encodeRequest(&badAdmissionReview))
+	myr := strings.NewReader(requestString)
+	r, _ := http.Post(server.URL, "application/json", myr)
+	review := decodeResponse(r.Body)
+	t.Log(review.Response)
+	if review.Response.Allowed {
+		t.Error("Allowed pipelinerun that should not have been allowed!")
+	}
```

Comment:
hmm... can we validate that it's complaining about the parameter not being valid and not something else? (currently that's what's expected and what happens, but if we add any other validation that might not be the case anymore and this test would need changing).

Reply:
We can, done